src/Core/Services/Implementations/CipherService.cs

```diff
@@ -222,10 +222,11 @@ public class CipherService : ICipherService
                 AttachmentId = attachmentId,
                 FileName = fileName,
                 Key = key,
-                Size = stream.Length
             };
 
             await _attachmentStorageService.UploadNewAttachmentAsync(stream, cipher, data);
+            // Must read stream length after it has been saved, otherwise it's 0
+            data.Size = stream.Length;
```

Comment:
To clarify - we can move the `size` declaration out of the initial upload because we update the same attachment data below in the `try/catch` with the correct size attribute?

Reply:
The initial data is actually not saved until the try/catch.

I moved it out of initialization because the value of `stream.Length` is 0 always. It will be the correct value any time after `await _attachmentStorageService.UploadNewAttachmentAsync(stream, cipher, data);`

The attachment data is saved at line 242.

Reply:
After digging a bit deeper into `_attachmentStorageService.UploadNewAttachmentAsync(...)` method it appears the only thing being used the the `data` is the `AttachmentId`, so what you're saying makes sense to me. :thumbsup: